--- netkit-rusers-0.17/rpc.rstatd/rstat_proc.c.old	2024-02-12 12:35:20.618002537 +0100
+++ netkit-rusers-0.17/rpc.rstatd/rstat_proc.c	2024-02-12 12:35:46.315174441 +0100
@@ -1082,7 +1082,7 @@
 
 	switch (rqstp->rq_proc) {
 	case NULLPROC:
-		(void)svc_sendreply(transp, xdr_void, (char *)NULL);
+		(void)svc_sendreply(transp, (xdrproc_t) xdr_void, (char *)NULL);
 		return;
 		break;
 
--- netkit-rusers-0.17/rpc.rstatd/rstatd.c.old	2024-02-12 12:47:37.635932927 +0100
+++ netkit-rusers-0.17/rpc.rstatd/rstatd.c	2024-02-12 12:48:04.091109903 +0100
@@ -77,7 +77,7 @@
 	 * See if inetd started us
 	 */
 	fromlen = sizeof(from);
-	if (getsockname(0, (struct sockaddr *)&from, &fromlen) < 0) {
+	if (getsockname(0, (struct sockaddr *)&from, (socklen_t *)&fromlen) < 0) {
 		from_inetd = 0;
 		sock = RPC_ANYSOCK;
 		proto = IPPROTO_UDP;
